### DISCUSSION TITLE
I added a download plugin, how to run my custom function when I click the download button?

### DISCUSSION THREAD

**Discussion by tolyle** [igordanchenko/yet-another-react-lightbox#111] [Q&A] [answered]
Hi,

I want to run a custom function when I click the download button, and then download it after some code in this function, for example, I need to save client ip, time, photoName and so on to my database. but I do not know how to achieve this function , althought I read the documents many times.

Thank you.

```
 <Lightbox
        controller={{ ref }}
        slides={slides}
        open={index >= 0}
        index={index}
        close={() => setIndex(-1)}
        on={{
          click: () => {
            alert('a');
            //save client IP, photoName .....
            // download photo
          },
        }}
        // enable optional lightbox plugins
        plugins={[Fullscreen, Slideshow, Zoom, Captions, Download]}
      />
```

**Comment by igordanchenko** [comment]
I've added `on.download` callback in v3.5.1

```jsx
<Lightbox
  on={{
    download: ({ index }) => {
      console.log("Download", index);
    },
  }}
  // ...
/>
```

**Reply by tolyle** [reply]
Thank you very much, you add a callback function, but it does not seem to have fully realized the function I want.  Althought I write return code into the callback function,it can still be downloaded,  I want to complete the download code by myself in the callback function. Should I how to do? In addition, when the slideshow is playing, I want to add a callback when the play or pause button is clicked, so that I can play music while the slideshow is playing, and when the user clicks pause, my music can also be Pause through callback function.

**Reply by tolyle** [reply]
```
 on={{
          download: ({ index }) => {
            console.log('Download', index);
            //download photo
            window.location.href = '';
            return;
          },
```

**Reply by igordanchenko** [reply, marked as answer]
> I want to complete the download code by myself in the callback function

If that's the case, and you do not want to use the Download plugin functionality, then you should probably implement your own Download button.

https://codesandbox.io/s/yet-another-react-lightbox-111-mbum69?file=/src/App.js

> In addition, when the slideshow is playing, I want to add a callback when the play or pause button is clicked, so that I can play music while the slideshow is playing, and when the user clicks pause, my music can also be Pause through callback function.

I'll add these callbacks in the next release.

**Reply by tolyle** [reply]
I got it, thank you and looking forward new version~

**Reply by igordanchenko** [reply]
Slideshow callbacks are now available in v3.5.2

**Reply by tolyle** [reply]
Thanks a lot.

**Reply by tolyle** [reply]
It seems that the new version has not been released to the npmjs .

```
PS D:\work\blog\blog> pnpm view yet-another-react-lightbox versions
[
  '1.0.0',      '1.0.1',      '1.1.0',      '1.1.1',      '1.1.2',
  '1.2.0',      '1.2.1',      '1.2.2',      '1.2.3',      '1.3.0',
  '1.3.1',      '1.3.2',      '1.3.3',      '1.3.4',      '1.3.5',
  '1.3.6',      '1.4.0',      '1.4.1',      '1.5.0',      '1.6.0',
  '1.7.0',      '1.7.1',      '1.8.0',      '1.8.1',      '1.9.0',
  '1.9.1',      '1.9.2',      '1.9.3',      '1.9.4',      '1.9.5',
  '1.9.6',      '1.10.0',     '1.11.0',     '1.11.1',     '1.12.0',
  '1.13.0',     '1.13.1',     '1.13.2',     '1.13.3',     '1.13.4',
  '1.14.0',     '2.0.0-rc.1', '2.0.0-rc.2', '2.0.0-rc.3', '2.0.0-rc.4',
  '2.0.0-rc.5', '2.0.0',      '2.0.1',      '2.0.2',      '2.0.3',
  '2.1.0',      '2.1.1',      '2.1.2',      '2.1.3',      '2.2.0',
  '2.2.1',      '2.2.2',      '2.2.3',      '2.2.4',      '2.2.5',
  '2.2.6',      '2.2.7',      '2.3.0',      '2.3.1',      '2.4.0',
  '2.4.1',      '2.4.2',      '2.4.3',      '2.5.0',      '2.5.1',
  '2.6.0',      '2.6.1',      '2.6.2',      '3.0.0',      '3.1.0',
  '3.2.0',      '3.3.0',      '3.4.0',      '3.5.0',      '3.5.1'
]
```

**Reply by igordanchenko** [reply]
Probably npm outage is still affecting your part of the world.

https://status.npmjs.org

<img width="590" alt="Screenshot 2023-04-25 at 6 27 47 AM" src="https://user-images.githubusercontent.com/64441155/234250091-d943fc8f-90e6-4785-80ea-d7095596cf22.png">

**Reply by tolyle** [reply]
OK,let me wait ：）

**Reply by igordanchenko** [reply]
Can you see the latest version on npm website? 

https://www.npmjs.com/package/yet-another-react-lightbox

<img width="598" alt="Screenshot 2023-04-25 at 8 43 53 AM" src="https://user-images.githubusercontent.com/64441155/234280274-36ec8f80-a47c-4004-a2ed-4dfc5efd09f6.png">

**Comment by tolyle** [comment]
Yes, I can see it on npm website .


获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Igor Danchenko ***@***.***>
发送时间: Tuesday, April 25, 2023 8:51:15 PM
收件人: igordanchenko/yet-another-react-lightbox ***@***.***>
抄送: tolyle ***@***.***>; Author ***@***.***>
主题: Re: [igordanchenko/yet-another-react-lightbox] I added a download plugin, how to run my custom function when I click the download button? (Discussion #111)


Can you see the latest version on npm website?

https://www.npmjs.com/package/yet-another-react-lightbox

[Screenshot 2023-04-25 at 8 43 53 AM]<https://user-images.githubusercontent.com/64441155/234280274-36ec8f80-a47c-4004-a2ed-4dfc5efd09f6.png>

―

**Reply by tolyle** [reply]
I have updated to the latest version 3.5.2, thank you.
